cmd/thermal-recorder/cptv_test.go

```
@@ -0,0 +1,150 @@
+		"skyline.cptv         Detected: None             Recorded: None             Motion frames: 1/91",
+	}
+
+	assert.Equal(t, expectedResults, results)
+}
+
+// DoTestResearchAnimalRecordings - change this to test to run though different scenarios of test
+// calculations.   It will output the results to /motiontest/results
+func DoTestResearchAnimalRecordings(t *testing.T) {
+	testname := "cut off - noise"
+	searchDir := GetBaseDir() + "/motiontest/noise"
+
+	f, err := os.Create(GetBaseDir() + "/motiontest/results/" + testname)
+	if err != nil {
+		return
+	}
+	defer f.Close()
+	writer := bufio.NewWriter(f)
+
+	config := CurrentConfig()
+	config.Motion.TempThresh = 2900
+	ExperiementAndWriteResultsToFile(testname+"2900", config, searchDir, writer)
+
+	config.Motion.TempThresh = 2800
+	ExperiementAndWriteResultsToFile(testname+"2800", config, searchDir, writer)
+
+	config.Motion.TempThresh = 2700
+	ExperiementAndWriteResultsToFile(testname+"2700", config, searchDir, writer)
+
+	config.Motion.TempThresh = 2500
+	ExperiementAndWriteResultsToFile(testname+"2500", config, searchDir, writer)
+
+	ExperiementAndWriteResultsToFile("Current config", CurrentConfig(), searchDir, writer)
+
+	ExperiementAndWriteResultsToFile("Old default", OldDefaultConfig(), searchDir, writer)
+}
+
+func ExperiementAndWriteResultsToFile(name string, config *Config, dir string, writer *bufio.Writer) {
+	fmt.Fprintf(writer, "Results for %s", name)
+	fmt.Fprintln(writer)
+
+	results := NewCPTVPlaybackTester(config).TestAllCPTVFiles(dir)
+
+	fmt.Fprintf(writer, "%-10s:  %ds - %ds", "Recording limits", config.MinSecs, config.MaxSecs)
+	fmt.Fprintln(writer, "")
+	fmt.Fprintln(writer, config.Motion)
+
+	for ii := range results {
+		fmt.Fprintln(writer, results[ii])
+	}
+	fmt.Fprintln(writer)
+	fmt.Fprintln(writer)
+
+	writer.Flush()
+}
+
+func GetBaseDir() string {
+	_, file, _, _ := runtime.Caller(0)
+
+	dir, _ := filepath.Abs(filepath.Dir(file))
```

Again, don't ignore the error, panic if it happens. Panics are rarely a good idea for production code but are fine in test code.